components/layout_2020/flow/construct.rs

```diff
@@ -591,7 +591,9 @@ where
 
         let kind = match contents.try_into() {
             Ok(contents) => match display_inside {
-                DisplayInside::Flow { is_list_item } => {
+                DisplayInside::Flow { is_list_item }
+                    if !info.style.get_box().overflow_x.is_scrollable() =>
```

Comment:
I think this should be a new function on the `StyleExt` crate, because there are [quite a few properties](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Block_formatting_context) that trigger the creation of a new formatting context and we should try to handle them all. In addition, I think this should check both `overflow_x` and `overflow_y`.

Reply:
I agree with turning this into its own function.

Checking `overflow-y` isn't probably necessary since elements can't be a scroll container in one axis but not in the other. The style adjuster takes care of changing the other property.

However, not all elements with `overflow: scroll` end up being scroll containers (e.g. replaced elements). So maybe it would be better to check layout rather than the computed style. Not sure if it matters in practice, though.

Reply:
I'm not very sure what the function should have since establishing formatting context is quite a complex topic and this check only happens when the element is a block, or, have `DisplayOutside::Block` and `DisplayInside::Flow`. Some is handled in other places(floats, abs-pos elements, `<html>`), Some is other kind of formatting context(Flexbox), and Some are not actually implemented(tables, multicol, grid).

Reply:
> I'm not very sure what the function should have since establishing formatting context is quite a complex topic and this check only happens when the element is a block, or, have `DisplayOutside::Block` and `DisplayInside::Flow`. Some is handled in other places(floats, abs-pos elements, `<html>`), Some is other kind of formatting context(Flexbox), and Some are not actually implemented(tables, multicol, grid).

It's true that not all situations would be useful to put in this function, but offhand this could also check whether the contents are `display: flow-root` when we add support for that as well as `column-span: all`. In short, I think there will need to be more situations in the not too distant future.

Reply:
> However, not all elements with `overflow: scroll` end up being scroll containers (e.g. replaced elements). So maybe it would be better to check layout rather than the computed style. Not sure if it matters in practice, though.

Since spec says bfc is established when `overflow` is not `visible` or `clip`, IMO this is OK.

Reply:
Ah, that's right, it checks the compute value.